apps/studio/src/server/modules/page/page.router.ts

```diff
@@ -257,8 +258,19 @@ export const pageRouter = router({
         pageId,
         userId: ctx.user.id,
       })
-      return addedVersionResult
 
-      /* TODO: Step 2: Use AWS SDK to start a CodeBuild */
+      /* Step 2: Use AWS SDK to start a CodeBuild */
+      const site = await getSiteNameAndCodeBuildId(siteId)
+      const codeBuildId = site.codeBuildId
+      if (!codeBuildId) {
+        throw new TRPCError({
```

Comment:
To fix: Somehow though error is thrown, the FE's PublishButton's mutation's onError handlers do not kick in. Because of that, the button is in a perpetual loading state. Any idea why?

Reply:
let me take a look at this

Reply:
thanks @seaerchin

Reply:
Fixed - need to disable retry on failure for mutations